src/components/linterlib/chktex.ts

```diff
@@ -0,0 +1,307 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as os from 'os'
+
+import type {Extension} from '../../main'
+import { LinterUtil } from './linterutil'
+import { convertFilenameEncoding } from '../../utils/convertfilename'
+
+export class ChkTeX {
+    readonly #linterName = 'ChkTeX'
+    readonly linterDiagnostics: vscode.DiagnosticCollection = vscode.languages.createDiagnosticCollection(this.#linterName)
+    readonly #linterUtil: LinterUtil
+
+    constructor(private readonly extension: Extension) {
+        this.#linterUtil = new LinterUtil(extension)
+    }
+
+    async lintRootFile() {
```

Comment:
The methods `lintRootFile` and `lintFile` share a lot of common code. Actually, they only differ by a couple of lines. I suggest to refactor these two methods to extract the command part to ease maintainance.